### PR TITLE
Form built with form_for and other helper methods

### PR DESCRIPTION
**Pull request by Bekhzod96**
**Created a form using different helper method which provided HTML and Rails.**

First, we created form by simple HTML form tag. 
- In this step, it was needed to authenticate with extra HTML tag

Second, we created by form_tag rails helper 
- Here form_tag was useful to authenticate but required to refactor data and specify the router  

Finally, we created form_for rails helper 
- finally, the helper does its job and according to convention it defines all path,  data handling and authenticates by rails convention

If you have some suggestions feel free to create an issue

**Thanks for recommendation and suggestions**

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+    # @user = User.new(username: params[:username], email: params[:email], password: params[:password])
```
Comment (shshamim63) [inline app/controllers/users_controller.rb]: Please remove the commented code.

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    else
+      render 'edit'
+    end
+  end
+
+  private
+
+  def user_params
+    params.require(:user).permit(:username, :email, :password)
+  end
```
Comment (shshamim63) [inline app/controllers/users_controller.rb:32]: Wow, you know the basic rules. Amazing.

**Review by shshamim63 — CHANGES_REQUESTED** [review]
## Status :recycle: Required Changes :recycle: 
___
Hi @Bekhzod96 and @javierbotero :wave: 
Awesome job :clap: You have covered most of the tasks. But there are some issues that are blocking me to approve this project right now. Let's resolve them,
___
### :spades: Mandatory Changes :spades: 
___
- I can not see a block in your new.html.erb file related to <form> tag. This is the first stage from where it mutates to the final form for.
image
Please add it.
2. Please remove the commented code. from the controller.
3. The title of the Pull Request is not self-descriptive. Please have it meaningful.
4. Are you sure just rails s will allow me to perform task? Cause there is migration. Don't I require to run migrate and create a database? If I need to do them then the info in the readme related to getting started is not enough. Please add the rest info.
___
#### Please fix the above-mentioned issue and send another review request.
___
## :herb: Happy Coding :herb: 
### Shakhawat Hossain

**Review by uche-inyama — CHANGES_REQUESTED** [review]
## Changes Required

Hi team :wave: 

Good job so far, however, there is a detail that needs modification in your readme.

In your readme, where you mentioned how to get the program running, it should follow this order 

-  Clone the repository.
-  cd into the folder.
- run bundle install
- run rails db:migrate, then
- run rails s.

Update your README file with this sequence, then submit a form for a code review.


#### ``` Have a great day ``` :slightly_smiling_face: 

``` Uchechukwu Henry Inyama ```

**Review by Ispirett — APPROVED** [review]
## Status Approved :1st_place_medal: 
Great job folks you have done well with all the changes made you can move on now.  :+1:

**Pull request merged**